packages/map-template/src/components/Search/Categories/Categories.jsx

```diff
@@ -141,29 +153,61 @@ function Categories({ onSetSize, getFilteredLocations, searchFieldRef, isOpen })
         }
     }, [activeCategory, category, isBottomSheetLoaded]);
 
-    /**
-     * Collects all child keys from all categories.
-     * Filters out categories from the initial view that are listed as a child to another category.
+    /*
+     * Filters and sets the categories to display based on whether top-level or sub-categories should be shown.
+     * If `topLevelCategory` is true, it excludes categories that appear as children.
+     * Otherwise, it includes only categories that are children.
      */
     useEffect(() => {
         const childKeys = categories.flatMap(([, category]) => category.childKeys || []);
-        const topLevelCategories = categories.filter(([key]) => !childKeys.includes(key));
-        setCategoriesWithoutChildKeys(topLevelCategories);
-    }, [categories]);
+        const categoriesToDisplay = topLevelCategory ? categories.filter(([key]) => !childKeys.includes(key)) : categories.filter(([key]) => childKeys.includes(key));
+        setCategoriesToShow(categoriesToDisplay)
+    }, [categories])
+
+    /*
+     * Gets display name of selected category.
+     */
+    useEffect(() => {
+        const categoryDisplayName = categories.find(([key]) => key === selectedCategory)?.[1]?.displayName;
+        setSelectedCategoryDisplayName(categoryDisplayName)
+    }, [categories])
 
     return (
         <div className="categories prevent-scroll" {...scrollableContentSwipePrevent}>
             {categories.length > 0 && (
                 <div className="categories__list">
-                    {categoriesWithoutChildKeys
-                        .map(([category, categoryInfo]) => (
+                    {topLevelCategory ? (
+                        categoriesToShow.map(([category, categoryInfo]) => (
                             <div key={category} className="categories__category">
                                 <button onClick={() => categoryClicked(category)}>
                                     <img src={categoryInfo.iconUrl} alt="" />
                                     {categoryInfo.displayName}
                                 </button>
                             </div>
-                        ))}
+                        ))
+                    ) : (
+                        <>
+                            <div className="categories__nav">
+                                <button
+                                    aria-label={t('Back')}
+                                    type="button"
+                                    className="categories__nav--button"
+                                    onClick={handleBack}>
+                                    <ChevronLeft />
+                                </button>
+                                <div>{selectedCategoryDisplayName}</div>
+                            </div>
+                            {selectedCategoriesArray.current.length === 1 &&
+                                categoriesToShow.map(([category, categoryInfo]) => (
```

Comment:
Could we change the JSX so we don't repeat the `{categoriesToShow.map(([category, categoryInfo]) => ( <div /> ))}` part?
I imagine the `<div className="categories__nav" />` could be rendered conditionally based on the topLevelCategory prop.